internal/config/auth_config_test.go

```diff
@@ -0,0 +1,302 @@
+package config
```

Comment:
Note that I have not written tests for `KnownHosts` (because it has no extra behaviour on top of `go-gh/auth` right now so it was low priority) nor the `AliasConfig` because it is low risk for the multi account changes.

Reply:
I have written tests for `Hosts` and `DefaultHost` now but I will leave `AliasConfig` out of this PR and do it in a follow up since this is already getting pretty hefty.